src/protocols/rdp/rdp_fs.c

```diff
@@ -38,7 +38,11 @@
 
 #include <guacamole/pool.h>
 
-guac_rdp_fs* guac_rdp_fs_alloc(guac_client* client, const char* drive_path) {
+guac_rdp_fs* guac_rdp_fs_alloc(guac_client* client, const char* drive_path, int create_drive_path) {
+
+	if(create_drive_path) {
+		mkdir(drive_path, S_IRWXU);
```

Comment:
The return value of ```mkdir()``` needs to be checked. If an error occurs, some useful message should be logged via ```guac_client_log()```. I'm guessing the flags given may need to be adjusted, too, as ```mkdir()``` will fail if the directory already exists. Perhaps this was your intent ... but the other ways ```mkdir()``` can fail are important and should be logged.

I'd also suggest including a debug-level log message just noting that the directory is being created, since that is fairly useful information.